app/controllers/maintenance_tasks/tasks_controller.rb

```diff
@@ -38,6 +39,12 @@ def run
 
     private
 
+    def task_arguments
+      return {} unless params[:task_arguments].present?
+      task_attributes = Task.named(params[:id]).attribute_names
```

Comment:
Previously a wrong Task name would result in a `ActiveRecord::RecordInvalid` raised from Runner and error caught in the action, now it will raise here and will result in a 500. We don't have tests for this because we didn't want controller tests and there's no way to test this with the UI (with a browser test, you'd need to see the UI and remove the task before clicking on Run). We could ignore that for now because like I mentioned in a previous review, parts of this logic should move to the Runner, with the CLI job needing similar handling, and fix it once we support arguments from the CLI.

Reply:
Hum, interesting. Let's leave it for now, I can remove it in the CLI PR if we choose to go that route and we feel it's redundant. Alternatively, we could keep the use of the Strong Parameters API here and just rescue `Task::NotFoundError` and return `{}`. I'm suspecting that we may just want to check the keys in `validate_task_arguments` though, since we're already rescuing on `Task::NotFoundError` there. But I'll defer to the CLI PR.